src/stats.rs

```diff
@@ -287,6 +295,89 @@ impl Reporter {
     }
 }
 
+trait SendStat {
+    fn send_increment(&self, name: &str);
+
+    fn send_count(&self, name: &str, count: i64);
+
+    fn send_time(&self, name: &str, time: u64);
+
+    fn send_gauge(&self, name: &str, value: u64);
+
+    fn submit_count_stat(&self, metric_builder: MetricBuilder<cadence::Counter>) {
+        // TODO: Move tagging logic here
+        if metric_builder.try_send().is_err() {
+            error!("Error sending query metrics to client");
+        }
+    }
+}
+
+impl SendStat for StatsdClient {
+    fn send_increment(&self, name: &str) {
+        let metric_builder: MetricBuilder<cadence::Counter> = self.incr_with_tags(name);
+        self.submit_count_stat(metric_builder);
+    }
+
+    fn send_count(&self, name: &str, count: i64) {
+        let metric_builder = self.count_with_tags(name, count);
+        self.submit_count_stat(metric_builder);
+    }
+
+    fn send_time(&self, name: &str, time: u64) {
+        let metric_builder = self.time_with_tags(name, time);
+        if metric_builder.try_send().is_err() {
+            error!("Error sending query metrics to client");
+        }
+    }
+
+    fn send_gauge(&self, name: &str, value: u64) {
+        let metric_builder = self.gauge_with_tags(name, value);
+        if metric_builder.try_send().is_err() {
+            error!("Error sending query metrics to client");
+        }
+    }
+}
+
+fn new_statsd_client() -> StatsdClient {
+    let config = get_config();
+    if config.general.enable_statsd {
+        let statsd_prefix =
+            env::var("STATSD_PREFIX").expect("Missing STATSD_PREFIX environment variable");
+
+        // Prefer to use socket over udp
+        let sink = match env::var("STATSD_SOCKET") {
+            Ok(statsd_sock) => {
+                let socket = UnixDatagram::unbound().unwrap();
+                socket.set_nonblocking(true).unwrap();
+                let buffered_sink = BufferedUnixMetricSink::from(statsd_sock, socket);
+                QueuingMetricSink::from(buffered_sink)
+            }
+            Err(_) => {
+                let statsd_host =
```

Comment:
Why not use the toml config?

Reply:
As far as I know, environment variables are the more standard way to configure statsd clients. Also didn't wanna add a bunch of conditional things in the config logic to deal with the different types of clients (unix socket vs udp)

Reply:
I agree with Lev on this one. I think we should stick to one form of configuration